src/jdk.incubator.vector/share/classes/jdk/incubator/vector/Float16.java

```diff
@@ -1401,8 +1412,15 @@ public static Float16 fma(Float16 a, Float16 b, Float16 c) {
         // product is numerically exact in float before the cast to
         // double; not necessary to widen to double before the
         // multiply.
-        double product = (double)(a.floatValue() * b.floatValue());
-        return valueOf(product + c.doubleValue());
+        short fa = float16ToRawShortBits(a);
```

Comment:
The new implementations in fma and sqrt are comparatively long and obscure compared to the current versions. That might be the price of intrinsification, but it would be helpful to at least have a comment to the reader explaining why the more obvious code was not being used.

Reply:
@jatin-bhateja could we change the intrinsic to declare the three Float16 values as additional parameters which are only ever passed to the lambda? I believe when intrinsic we will just drop those extra parameters.

Reply:
Our intent here is to explicitly unbox the float16 values in Java code to avoid complexifying unboxing by C2. In that case, the VM needs to be aware of the Float16 class and record its field offsets, as it did for the [VectorPayload class.](https://github.com/openjdk/jdk/blob/master/src/hotspot/share/classfile/javaClasses.cpp#L5065) 

I have re-structured the code to remove unnecessary obfuscation introduced as a side effect of intrinsification. Let me know if we need to refine it further.